### PR TITLE
makes the fileserver configurable per service

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- makes the fileserver configurable per service

## Why are we making these changes?

We wish to make the configurable per service. For example, this can be useful while trying to support limited access to backend instance logs.

### CODE REVIEW

File: waiter/config-full.edn [hunk 1]
```diff
@@ -482,6 +482,9 @@
                                                            :default-container-image "twosigma/waiter-test-apps:latest"
                                                            ;; The default namespace to use for Kubernetes objects created by Waiter
                                                            :default-namespace "waiter"
+                                                           ;; Function that determines if the fileserver container should be enabled for a service.
+                                                           ;; The function takes the service description as input.
+                                                           :fileserver-container-enabled? waiter.some.package/fileserver-enabled?
```
Comment (DaoWen) [inline waiter/config-full.edn:487]: I strongly prefer that this be implemented as a `:predicate-fn` inside the `:fileserver` config like we did with the `:reverse-proxy` for consistency.

Comment (DaoWen) [reply waiter/config-full.edn:487]: We would also want this function to have the same signature as the `:reverse-proxy :predicate-fn` for consistency and flexibility.

Comment (shamsimam) [reply waiter/config-full.edn:487]: sgtm, will change

**Review by shamsimam — COMMENTED** [review]
Addressed requested changes

**Review by DaoWen — APPROVED** [review]

**Pull request merged**